File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
//        set counter:
        int counter = 0;
//        loop through nums:
        for (int i=0; i<nums.size(); i++) {
//            loop through nums again:
            for (int j=i+1; j<nums.size(); j++) {
//                if match, increment counter
                if (nums[i] == nums[j]) {
                    counter++;
                } else {
                    continue;
                }
            }
        }
        return counter;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
//       solving with hashmap:
        HashMap<Integer, Integer> map = new HashMap<>();
        int count = 0;
//         loop through nums:
        for (int i = 0; i < nums.length; i++) {
//             if map contains this num
            if (map.containsKey(nums[i])) {
//                 increment counter first with value of number in map !important
                count += map.get(nums[i]);
                System.out.println(map.get(nums[i]));
                map.put(nums[i], map.get(nums[i]) + 1);
            } else {
                map.put(nums[i], 1);
            }
        }
        return count;
    }
}